internal/pkg/template/repository/manager.go

```diff
@@ -17,38 +19,45 @@ const OperationTimeout = 30 * time.Second
 type Manager struct {
 	ctx                 context.Context
 	logger              log.Logger
-	lock                *sync.Mutex
 	defaultRepositories []model.TemplateRepository
+	initGroup           *singleflight.Group
```

Comment:
Upravil som metodu `Repository` (resp. `AddRepository`) aby nepouzivala zamok `lock`.
Tj. parallelne vieme pridat viacero repositories (pri starte serveru).

No musime zabezpecit, aby bezala iba jedna inicializacia naraz.
Nato je dobra utilitka `sync.Once`, ale iba pre jednu hodnotu.
My to chceme samostatne pre kazdy `hash` repository.

Na to je dobra utilitka `singleflight` - `provides a duplicate function call suppression mechanism.`:
https://pkg.go.dev/golang.org/x/sync/singleflight

Vybral som ju preto, lebo prave podporuje `keys`, tj. viac hodnot a tak je nas kod jednoduchsi.

Reply:
BTW. balicky z `/x/` su oficialne balicky, ktore vznikli nedavno (rozumej par rokov dozadu :slightly_smiling_face: ) a este neboli zaradene do standardnej kniznice .... tj. kvalitou su medzi standardnou kniznicou a 3rd-party.